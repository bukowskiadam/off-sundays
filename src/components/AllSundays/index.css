.row {
    display: flex;
    max-width: 400px;
    margin: auto;
}

.row.closed {
    color: #da0000;
}

.row.opened {
    color: #8b8b8b;
}

.row .date {
    flex-grow: 1;
    flex-basis: 60%;
}

.row .value {
    flex-grow: 1;
    flex-basis: 40%;
}
