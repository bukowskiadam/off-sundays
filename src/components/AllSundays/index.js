import React from 'react';
import PropTypes from 'prop-types';

import { allSundays } from '../../helpers/dateOperations';
import { isClosed } from '../../data';

import DayList from '../DayList';
import BoxContainer from '../BoxContainer';

const AllSundays = ({ year }) => (
    <div>
        <h3>{year}</h3>
        <BoxContainer>
            {allSundays(year).map(date => <DayList key={date.toString()} date={date} isClosed={isClosed(date)} />)}
        </BoxContainer>
    </div>
);

AllSundays.propTypes = {
    year: PropTypes.number.isRequired,
};

export default AllSundays;
