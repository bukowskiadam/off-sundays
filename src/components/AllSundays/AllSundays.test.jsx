import React from 'react';
import { shallow } from 'enzyme';

import AllSundays from './AllSundays';

describe('AllSundays', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<AllSundays year={2018}/>);

        expect(wrapper).toMatchSnapshot();
    });
});
