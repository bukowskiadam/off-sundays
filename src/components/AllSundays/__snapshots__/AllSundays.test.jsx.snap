// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllSundays renders correctly 1`] = `
<div>
  <h3>
    2018
  </h3>
  <BoxContainer>
    <Day
      date={2018-01-07T00:00:00.000Z}
      isClosed={false}
      key="1515283200000"
    />
    <Day
      date={2018-01-14T00:00:00.000Z}
      isClosed={false}
      key="1515888000000"
    />
    <Day
      date={2018-01-21T00:00:00.000Z}
      isClosed={false}
      key="1516492800000"
    />
    <Day
      date={2018-01-28T00:00:00.000Z}
      isClosed={false}
      key="1517097600000"
    />
    <Day
      date={2018-02-04T00:00:00.000Z}
      isClosed={false}
      key="1517702400000"
    />
    <Day
      date={2018-02-11T00:00:00.000Z}
      isClosed={false}
      key="1518307200000"
    />
    <Day
      date={2018-02-18T00:00:00.000Z}
      isClosed={false}
      key="1518912000000"
    />
    <Day
      date={2018-02-25T00:00:00.000Z}
      isClosed={false}
      key="1519516800000"
    />
    <Day
      date={2018-03-04T00:00:00.000Z}
      isClosed={false}
      key="1520121600000"
    />
    <Day
      date={2018-03-11T00:00:00.000Z}
      isClosed={true}
      key="1520726400000"
    />
    <Day
      date={2018-03-18T00:00:00.000Z}
      isClosed={true}
      key="1521331200000"
    />
    <Day
      date={2018-03-25T00:00:00.000Z}
      isClosed={false}
      key="1521936000000"
    />
    <Day
      date={2018-03-31T23:00:00.000Z}
      isClosed={true}
      key="1522537200000"
    />
    <Day
      date={2018-04-07T23:00:00.000Z}
      isClosed={true}
      key="1523142000000"
    />
    <Day
      date={2018-04-14T23:00:00.000Z}
      isClosed={true}
      key="1523746800000"
    />
    <Day
      date={2018-04-21T23:00:00.000Z}
      isClosed={true}
      key="1524351600000"
    />
    <Day
      date={2018-04-28T23:00:00.000Z}
      isClosed={false}
      key="1524956400000"
    />
    <Day
      date={2018-05-05T23:00:00.000Z}
      isClosed={false}
      key="1525561200000"
    />
    <Day
      date={2018-05-12T23:00:00.000Z}
      isClosed={true}
      key="1526166000000"
    />
    <Day
      date={2018-05-19T23:00:00.000Z}
      isClosed={true}
      key="1526770800000"
    />
    <Day
      date={2018-05-26T23:00:00.000Z}
      isClosed={false}
      key="1527375600000"
    />
    <Day
      date={2018-06-02T23:00:00.000Z}
      isClosed={false}
      key="1527980400000"
    />
    <Day
      date={2018-06-09T23:00:00.000Z}
      isClosed={true}
      key="1528585200000"
    />
    <Day
      date={2018-06-16T23:00:00.000Z}
      isClosed={true}
      key="1529190000000"
    />
    <Day
      date={2018-06-23T23:00:00.000Z}
      isClosed={false}
      key="1529794800000"
    />
    <Day
      date={2018-06-30T23:00:00.000Z}
      isClosed={false}
      key="1530399600000"
    />
    <Day
      date={2018-07-07T23:00:00.000Z}
      isClosed={true}
      key="1531004400000"
    />
    <Day
      date={2018-07-14T23:00:00.000Z}
      isClosed={true}
      key="1531609200000"
    />
    <Day
      date={2018-07-21T23:00:00.000Z}
      isClosed={true}
      key="1532214000000"
    />
    <Day
      date={2018-07-28T23:00:00.000Z}
      isClosed={false}
      key="1532818800000"
    />
    <Day
      date={2018-08-04T23:00:00.000Z}
      isClosed={false}
      key="1533423600000"
    />
    <Day
      date={2018-08-11T23:00:00.000Z}
      isClosed={true}
      key="1534028400000"
    />
    <Day
      date={2018-08-18T23:00:00.000Z}
      isClosed={true}
      key="1534633200000"
    />
    <Day
      date={2018-08-25T23:00:00.000Z}
      isClosed={false}
      key="1535238000000"
    />
    <Day
      date={2018-09-01T23:00:00.000Z}
      isClosed={false}
      key="1535842800000"
    />
    <Day
      date={2018-09-08T23:00:00.000Z}
      isClosed={true}
      key="1536447600000"
    />
    <Day
      date={2018-09-15T23:00:00.000Z}
      isClosed={true}
      key="1537052400000"
    />
    <Day
      date={2018-09-22T23:00:00.000Z}
      isClosed={true}
      key="1537657200000"
    />
    <Day
      date={2018-09-29T23:00:00.000Z}
      isClosed={false}
      key="1538262000000"
    />
    <Day
      date={2018-10-06T23:00:00.000Z}
      isClosed={false}
      key="1538866800000"
    />
    <Day
      date={2018-10-13T23:00:00.000Z}
      isClosed={true}
      key="1539471600000"
    />
    <Day
      date={2018-10-20T23:00:00.000Z}
      isClosed={true}
      key="1540076400000"
    />
    <Day
      date={2018-10-27T23:00:00.000Z}
      isClosed={false}
      key="1540681200000"
    />
    <Day
      date={2018-11-04T00:00:00.000Z}
      isClosed={false}
      key="1541289600000"
    />
    <Day
      date={2018-11-11T00:00:00.000Z}
      isClosed={true}
      key="1541894400000"
    />
    <Day
      date={2018-11-18T00:00:00.000Z}
      isClosed={true}
      key="1542499200000"
    />
    <Day
      date={2018-11-25T00:00:00.000Z}
      isClosed={false}
      key="1543104000000"
    />
    <Day
      date={2018-12-02T00:00:00.000Z}
      isClosed={false}
      key="1543708800000"
    />
    <Day
      date={2018-12-09T00:00:00.000Z}
      isClosed={true}
      key="1544313600000"
    />
    <Day
      date={2018-12-16T00:00:00.000Z}
      isClosed={false}
      key="1544918400000"
    />
    <Day
      date={2018-12-23T00:00:00.000Z}
      isClosed={false}
      key="1545523200000"
    />
    <Day
      date={2018-12-30T00:00:00.000Z}
      isClosed={false}
      key="1546128000000"
    />
  </BoxContainer>
</div>
`;
